packages/flutter_tools/lib/src/analyze_project.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+enum StatusProjectValidator {
+  error,
+  warning,
+  success,
+}
+
+class ProjectValidatorResult {
+
+  ProjectValidatorResult(this.name, this.value, this._status, {this.warning});
+
+  final String name;
+  final String value;
+  final String? warning;
+  final StatusProjectValidator _status;
+
+  StatusProjectValidator get status{
+    return _status;
+  }
+
+  @override
+  String toString() {
+    if (_status == StatusProjectValidator.error) {
```

Comment:
if you are returning early, you can un-nest the conditionals to make reading this easier:

```dart
    if (_status == StatusProjectValidator.error) {
      return 'Error: $value';
    }
    String resultString = '$name: $value';
    if (warning != null) {
      return '$name: $value (warning: $warning)';
    }
    return '$name: $value';
  }
```